src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -28,7 +28,7 @@
  *
  * Parameter and service keys are case insensitive.
  *
- * A service id can contain lowercased letters, digits, underscores, and dots.
+ * A service id can contain letters, digits, underscores, and dots.
```

Comment:
My guess is this was written from a file loader pov, As in fact any character is allowed as of 2.7.